src/ncp/posix/netif.cpp

```diff
@@ -87,6 +90,35 @@ void Netif::Deinit(void)
     Clear();
 }
 
+void Netif::Process(const MainloopContext *aContext)
+{
+    if (FD_ISSET(mTunFd, &aContext->mErrorFdSet))
+    {
+        close(mTunFd);
+        DieNow("Error on Tun Fd!");
+    }
+
+    if (FD_ISSET(mTunFd, &aContext->mReadFdSet))
+    {
+        ProcessIp6Send();
+    }
+}
+
+void Netif::UpdateFdSet(MainloopContext *aContext)
+{
+    assert(aContext != nullptr);
+    assert(mTunFd >= 0);
+    assert(mIpFd >= 0);
+
+    FD_SET(mTunFd, &aContext->mReadFdSet);
+    FD_SET(mTunFd, &aContext->mErrorFdSet);
+
+    if (mTunFd > aContext->mMaxFd)
+    {
+        aContext->mMaxFd = mTunFd;
+    }
```

Comment:
Not for this PR but a possible enhancement idea (for future PR)

I assume this pattern of adding certain `fd` to `MainloopContext` (updating `mReadFdSet` and `mErrorFdSet` and then also updating `mMaxFd`) might be done in several places in the code.

It could be helpful to add a method on `MainloopContext` for this. This would simplify the code and ensure we always update `mMaxFd` as well.

Reply:
This is great! I'll do that in a seperate PR.